pkg/authclient/authclient.go

```diff
@@ -59,6 +60,8 @@ func (a *AuthClient) Resolve(ctx context.Context, accessKeyID string, clientIP s
 	}
 	for {
 		resp, err := client.Do(req)
+		middleware.AddReqIDHeader(ctx, resp)
```

Comment:
I think you want to move this up to req creation and add the header there, not to `resp`.